File: my-folder/0102-binary-tree-level-order-traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        levels = []
        if not root:
            return levels
        self.dfs(root,levels, 0)
        return levels
    
    def dfs(self, node, levels, level):
        if len(levels) == level:
            levels.append([])
        levels[level].append(node.val)
        if node.left:
            self.dfs(node.left, levels, level + 1)
        if node.right:
            self.dfs(node.right, levels, level + 1)
    

        

File: my-folder/0102-binary-tree-level-order-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if(root == null){
            return res;
        }

        Queue<TreeNode> que = new LinkedList<>();
        que.offer(root);

        while(!que.isEmpty()){
            List<Integer> curr = new ArrayList<>();
            int size = que.size();
            for(int i = 0; i < size; i++){
                TreeNode currnode = que.poll();
                curr.add(currnode.val);
                if(currnode.left != null){
                    que.offer(currnode.left);
                }
                if(currnode.right != null){
                    que.offer(currnode.right);
                }
            }
            res.add(curr);
        }

        return res;
    }
}
